homeassistant/helpers/config_validation.py

```diff
@@ -911,6 +914,42 @@ def script_action(value: Any) -> dict:
     has_at_least_one_key(CONF_SERVICE, CONF_SERVICE_TEMPLATE),
 )
 
+
+def find_root_schema(node: Any) -> vol.Schema:
+    """Traverse schema to find the root validation dictionary."""
+    if isinstance(node, dict):
+        return vol.Schema(node)
+
+    if isinstance(node, vol.Schema):
+        return find_root_schema(node.schema)
+
+    if isinstance(node, vol.All):
+        return find_root_schema(node.validators[0])
+
+    return None
+
+
+def describable_schema(node: Any, parent: Any = None) -> vol.Schema:
+    """Traverse schema until a dictionary is found then extend with description schema."""
+    if isinstance(node, dict):
+        if parent:
+            return parent.extend(STEP_DESCRIPTION_SCHEMA)
+
+        return vol.Schema(node).extend(STEP_DESCRIPTION_SCHEMA)
+
+    if isinstance(node, vol.Schema):
+        return describable_schema(node.schema, node)
+
+    if isinstance(node, vol.All):
+        new_schema = describable_schema(node.validators[0], node)
+        validators = list(node.validators)
+        validators[0] = new_schema
+        node.validators = tuple(validators)
+        return node
+
+    return node
```

Comment:
This helper might be overkill, but since many of the schemas put their dictionaries in different positions in their object hierarchy, all the`.extend(STEP_DESCRIPTION_SCHEMA)` lines had to go in different places, making it a bit harder to reason about.

The helper allows us to wrap around the full schema object for consistency everywhere, and makes scenarios easier to test.